Snakefile

```diff
@@ -16,20 +21,131 @@ def get_samples() -> dict[str, Sample]:
+        # Latch's JIT replaces each Latch URI in the config with a path to the local file.
+        # The `_latchfiles` field contains a mapping of each config key to the original Latch URI.
+        latch_path = os.path.join(config["_latchfiles"]["genomes_dir"], reference_id, f"{reference_id}.fna")
+        fasta_path = prebuilt_path if latchfile_exists(latch_path) else new_path
+    else:
+        fasta_path = prebuilt_path if os.path.exists(prebuilt_path) else new_path
+
+    return fasta_path
+
+
+def reference_cache_destination(wildcards: Wildcards) -> str:
+    """
+    Construct a path to a reference FASTA in the configured reference genome directory.
 
+    This function should *only* be used when parametrizing the destination path in
+    `build_reference`. It returns a Latch URI when the workflow is executing on Latch, so the built
+    reference can be copied back to the standard location. If the function is used as an `input` to
+    any rule, things will break because rule inputs must be specified as relative local paths.
+    """
+    reference_id = wildcards.reference_id
 
+    if workflow_is_executing_on_latch():
+        cached_path = os.path.join(config["_latchfiles"]["genomes_dir"], reference_id, f"{reference_id}.fna")
+    else:
+        cached_path = os.path.join(config["genomes_dir"], reference_id, f"{reference_id}.fna")
+
+    return cached_path
+
+
+####################################################################################################
+# Rules
+####################################################################################################
 rule all:
     input:
         expand("results/print_reference_path/{sample}.txt", sample=get_samples()),
         "results/collect_reference_paths/collected_reference_paths.txt"
 
 
+rule build_reference:
+    """
+    Build a reference and copy it to the configured reference location.
+
+    This rule is executed conditionally. If a built reference already exists in the configured
+    reference genomes directory (`config['genomes_dir']`), the rule is **not** executed.
+    Otherwise, a new reference is built.
+
+    Output:
+        fasta: A newly constructed "reference".
+    
+    Params:
+        destination_path: The expected path to the reference in the configured reference genomes
+            directory. The built reference will be copied to this location after construction.
+        cp_cmd: The command to use to copy the reference to the reference genomes directory. If the
+            workflow is executing on Latch, the reference genomes directory is assumed to be on
+            Latch, and `latch cp` is used. Otherwise, `cp` is used to copy the file to a local
+            directory.
+    """
+    output:
+        fasta="results/build_reference/{reference_id}/{reference_id}.fna"
+    params:
+        destination_path=reference_cache_destination,
+        cp_cmd=lambda wildcards: "latch cp" if workflow_is_executing_on_latch() else "cp"
```

Comment:
If local, should it be a `ln -s` instead?

Reply:
I prefer the copy, so the user can nuke the `results` directory without losing the cached files. Could make this configurable though

Reply:
I'm pretty sure deleting a softlink won't delete the linked file FYI

Reply:
Right, but this is going the other way.

The Snakefile writes the output of this rule to `results/build_reference/{reference_id}/`. This rule would additionally copy the file to the expected reference location, say `ref/genomes/{reference_id}/`.

If you replaced the copy with a symlink, you'd still have the file at `results/build_reference/`, but a soft-link at `ref/genomes/`.

Nuking `results` wouldn't delete the softlink, it would delete the target file.